src/editor.c

```diff
@@ -2458,7 +2457,7 @@ void editor_insert_text_block(GeanyEditor *editor, const gchar *text, gint inser
 	gint line_start = sci_get_line_from_position(sci, insert_pos);
 	GString *buf;
 	const gchar *eol = editor_get_eol_char(editor);
-	gint idx;
+	gint idx = 0;
```

Comment:
I'm not 100% sure this is the Right Fix™, and I didn't test it thoroughly enough.  @kugel- you might know better than I do maybe